crates/orchestrator/src/network/node.rs

```diff
@@ -441,4 +593,376 @@ impl std::fmt::Debug for NetworkNode {
     }
 }
 
-// TODO: mock and impl unit tests
+// TODO: mock and impl more unit tests
+#[cfg(test)]
```

Comment:
Awesome! thanks for also cover this old _TODO_

Reply:
Ha, just covered log parsing, but I think it might be a good starting point to add more tests :)